File: binarysearch.io/November 2020/305_palindromic_tree.py
# class Tree:
#     def __init__(self, val, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def solve(self, root):
        
        order = []
        def inorder(root):
            if root is None:
                return 
            inorder(root.left)
            order.append(root.val)
            inorder(root.right)
        inorder(root)
        return order == order[::-1]

File: binarysearch.io/305_palindromic_tree.cpp
/**
 * class Tree {
 *     public:
 *         int val;
 *         Tree *left;
 *         Tree *right;
 * };
 */
bool solve(Tree* root) {
    vector<int> order;
    vector<Tree *> stack;
    Tree * curr;
    Tree * node;
    curr = root;
    while(stack.size() || curr!=NULL){
        while(curr){
            stack.push_back(curr);
            curr = curr->left;
        }
        node = stack[stack.size()-1];
        stack.pop_back();
        order.push_back(node->val);
        curr = node->right;
    }
    int n = order.size();
    for(int i = 0; i < n; i++){
        if(order[i] != order[n-i-1]) return false;
    }
    return true;
    
}